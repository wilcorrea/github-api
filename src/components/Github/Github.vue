<template>
  <div class="container">
    <h1>{{ title }}</h1>
    <GithubSearch
      @searchUser="searchUser"
    />
    <GithubCard
      :user="user"
    />
    <h3>{{ error }}</h3>
  </div>
</template>

<script>
import axios from 'axios';
import GithubSearch from '../Github-search/Github-search';
import GithubCard from '../Github-card/Github-card';

export default {
  name: 'Github',
  messages: {
    error: 'User not found',
  },
  data() {
    return {
      user: {},
      error: '',
      title: 'Welcome to github playground api :)',
    };
  },
  methods: {
    searchUser(search) {
      this.cleanErrorMessages();
      return axios
        .get(`https://api.github.com/users/${search}`)
        .then(this.onSuccess)
        .catch(this.onError);
    },
    onSuccess(response) {
      this.user = { ...response.data };
    },
    onError() {
      this.user = {};
      this.error = this.$options.messages.error;
    },
    cleanErrorMessages() {
      this.error = '';
    },
  },
  components: {
    GithubSearch,
    GithubCard,
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
input.search {
  border-radius: 3px;
}
</style>
