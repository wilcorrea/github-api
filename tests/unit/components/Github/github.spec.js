import { shallowMount } from '@vue/test-utils'
// noinspection NpmUsedModulesInstalled
import Github from 'src/components/Github/Github.vue'
import axios from 'axios'

jest.mock('axios')

describe('Test Github', function () {
  const wrapper = shallowMount(Github)

  // it is valid??
  // it('check data', () => {
  //   expect(typeof wrapper.vm.user).toBe('object')
  //   expect(wrapper.vm.error).toBe('')
  //   expect(typeof wrapper.vm.title).toBe('string')
  // })

  // test cleanErrorMessages
  it('test cleanErrorMessages', () => {
    wrapper.vm.error = 'error'
    wrapper.vm.cleanErrorMessages()
    expect(wrapper.vm.error).toBe('')
  })

  // search a valid user
  it('search a valid user', async () => {
    const login = 'wilcorrea'
    const response = { data: { login } }
    axios.get.mockResolvedValue(response)
    await wrapper.vm.searchUser(login)
    expect(wrapper.vm.user.login).toBe(login)
  })

  // search a invalid user
  it('search a invalid user', async () => {
    wrapper.vm.error = ''
    await wrapper.vm.onError()
    expect(wrapper.vm.error).toBe(wrapper.vm.$options.messages.error)
  })

  // create and compare snapshots
  it('should render content correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
