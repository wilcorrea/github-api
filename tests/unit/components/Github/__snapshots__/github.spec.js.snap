// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Github should render content correctly 1`] = `"<div class=\\"container\\"><h1>Welcome to github playground api :)</h1> <githubsearch-stub></githubsearch-stub> <githubcard-stub user=\\"[object Object]\\"></githubcard-stub> <h3>User not found</h3></div>"`;
